### PR TITLE
Bigquery magic

### PR DESCRIPTION
**Pull request by julioyildo**
## References

Support of IPython Magics for BigQuery. 
see : https://googleapis.dev/python/bigquery/latest/magics.html

## Code changes

Added the folder `/packages/jupyterlab-lsp/src/transclusions/ipython-bigquery`

## User-facing changes

Added colorization of SQL syntax on the following command:
```
%%bigquery 
SELECT * FROM <table_id>
```

## Chores

- [ ] linted
- [ ] tested
- [ ] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by julioyildo** [comment]
Hello @krassowski,

I have commented some tests code because IPython Magics for BigQuery  doesn't support:
`%bigquery` line magic, database connections, `persists` tags, etc.

I agree is a bit confusing to keep them in the code then. I will delete them and PR again. 

 Is it convenient for you? :)

**Comment by krassowski** [comment]
Thank you, yes it makes sense to remove unused code if that magic does not make use of these functions.

The build now passes, but there are still some stylistic things that prettier complains about:

```
$ prettier --check "**/*{.ts,.tsx,.js,.jsx,.css,.json,.md,.yml}"
Checking formatting...
[warn] packages/jupyterlab-lsp/src/transclusions/defaults.ts
[warn] packages/jupyterlab-lsp/src/transclusions/ipython-bigquery/extractors.spec.ts
[warn] packages/jupyterlab-lsp/src/transclusions/ipython-bigquery/extractors.ts
[warn] Code style issues found in the above file(s). Forgot to run Prettier?

```

**Comment by krassowski** [comment]
As for the other failures:

```
Run python -m pip check
python-language-server 0.11.1 requires configparser, which is not installed.
python-language-server 0.11.1 requires pydocstyle, which is not installed.
```

This seems like a very strange conda resolver issue. Current python-language-server version is 0.36.2, not sure why it started installing 0.11.1 (it might be due to jedi pin, however we do have it pinned too, so possibly something else).

**Comment by julioyildo** [comment]
> The build now passes, but there are still some stylistic things that prettier complains about:
> 
> ```
> $ prettier --check "**/*{.ts,.tsx,.js,.jsx,.css,.json,.md,.yml}"
> Checking formatting...
> [warn] packages/jupyterlab-lsp/src/transclusions/defaults.ts
> [warn] packages/jupyterlab-lsp/src/transclusions/ipython-bigquery/extractors.spec.ts
> [warn] packages/jupyterlab-lsp/src/transclusions/ipython-bigquery/extractors.ts
> [warn] Code style issues found in the above file(s). Forgot to run Prettier?
> ```

Hummm... I'm little bit confused. Do you have any ideas on how to solve this issue please ? 

Thank you !

**Comment by krassowski** [comment]
Try just running `jlpm lint` to fix the formatting (assuming that you have set up the project as described in the [CONTRIBUTING.md](https://github.com/krassowski/jupyterlab-lsp/blob/master/CONTRIBUTING.md)). Don't worry about the pip check - this is something we will look into.

**Comment by julioyildo** [comment]
> Try just running `jlpm lint` to fix the formatting (assuming that you have set up the project as described in the [CONTRIBUTING.md](https://github.com/krassowski/jupyterlab-lsp/blob/master/CONTRIBUTING.md)). Don't worry about the pip check - this is something we will look into.


Done! It seems that other errors have been reported : 

```
Run python -m pip check
  python -m pip check
  shell: C:\Program Files\Git\bin\bash.EXE -l {0}
  env:
    CACHE_EPOCH: 0
    JLPM_CMD: jlpm --ignore-optional --prefer-offline --frozen-lockfile
    PYTHONUNBUFFERED: 1
    ATEST_RETRIES: 3
    PIP_DISABLE_PIP_VERSION_CHECK: 1
    MAMBAFORGE_URL: https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge
    CONDA: C:\Miniconda3
    CONDA_PKGS_DIR: C:\Users\runneradmin\conda_pkgs_dir
autopep8 1.5.6 has requirement pycodestyle>=2.7.0, but you have pycodestyle 2.6.0.
```

**Comment by bollwyvl** [comment]
Closing/reopening to restart CI, now that it should get further than install...

File: packages/jupyterlab-lsp/src/transclusions/defaults.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 import { IPYTHON_SQL_TRANSCLUSIONS } from './ipython-sql';
+import { IPYTHON_BIGQUERY_TRANSCLUSIONS } from './ipython-bigquery';
 import { IPYTHON_TRANSCLUSIONS } from './ipython';
 
 export const DEFAULT_TRANSCLUSIONS: JupyterFrontEndPlugin<void>[] = [
   IPYTHON_RPY2_TRANSCLUSIONS,
   IPYTHON_SQL_TRANSCLUSIONS,
-  IPYTHON_TRANSCLUSIONS
+  IPYTHON_TRANSCLUSIONS,
+  IPYTHON_BIGQUERY_TRANSCLUSIONS
```
Comment (krassowski) [inline packages/jupyterlab-lsp/src/transclusions/defaults.ts:11]:
```suggestion
  IPYTHON_BIGQUERY_TRANSCLUSIONS,
  IPYTHON_TRANSCLUSIONS
```

I think that there was a thing with IPython substitutions including a catch-all regexp that has to be registered as the last one. Not sure if this is stlll the case, but probably safer to keep it in that order.

**Comment by krassowski** [comment]
Thank you @julioyildo for this contribution :tada:! It will be included in the upcoming release.

**Comment by HalfMillennium** [comment]
@krassowski @julioyildo Quick question, does this update only introduce syntax highlighting for BigQuery statements, or does also provide database-specific autocompletion?

**Comment by krassowski** [comment]
The auto-completion for bigquery won't be database-aware; for that we would need a BigQuery language server which can be configured to access the database (and I am not aware of existence of such).

**Comment by HalfMillennium** [comment]
Understood. I'll begin looking into developing this, thank you!

**Comment by HalfMillennium** [comment]
> The auto-completion for bigquery won't be database-aware; for that we would need a BigQuery language server which can be configured to access the database (and I am not aware of existence of such).

Actually, just testing out some of the syntax highlighting, do I need to have an LSP already explicitly defined? I've simply enabled the extension in my Jupyterlab instance (with the BigQuery magic enabled), and I can't see any highlighting on my queries.

**Pull request merged**